File: utilis.js
const fs = require('fs');
const path = require('path');
const PAIRS_FILE = path.join(__dirname, 'pairs.json');

// Config
const OWNER_NUMBER = "254792770219"; // Only owner can use delpair/listpair
const WHATSAPP_CHANNEL = "https://whatsapp.com/channel/0029Vb7UKYqHbFVCW3uGad0l";
const TELEGRAM_CONTACT = "https://t.me/Handsome_primis_killer_kent";

// Initialize pairs.json if not exists
if (!fs.existsSync(PAIRS_FILE)) {
    fs.writeFileSync(PAIRS_FILE, JSON.stringify({ pairs: [] }, null, 2));
}

module.exports = {

  // Initialize bot (replace with actual WhatsApp bot init)
  initBot: function(botInstance) {
    console.log("⛧ＰＲＩＭΞ⛧ ᛕΙᄂᄂΞＲ ⛧CЯΛSᕼΞЯ⛧ ɃЦ₲ ɃØŦ is running...");
    console.log("Powered by ⛧ＰＲＩΜΞ⛧ kîᄂᄂér ⛧ƘΞИŦ⛧");
    return botInstance;
  },

  // Load current pairs
  loadPairs: function() {
    const data = fs.readFileSync(PAIRS_FILE);
    return JSON.parse(data);
  },

  // Save pairs
  savePairs: function(data) {
    fs.writeFileSync(PAIRS_FILE, JSON.stringify(data, null, 2));
  },

  // Pair user
  pairUser: function(number) {
    const pairs = this.loadPairs();
    if (pairs.pairs.find(u => u.number === number)) return false;
    pairs.pairs.push({ number, paired: true });
    this.savePairs(pairs);
    return true;
  },

  // Delete all pairs (owner only)
  delPair: function(requesterNumber) {
    if (requesterNumber !== OWNER_NUMBER) return false;
    const pairs = this.loadPairs();
    pairs.pairs = [];
    this.savePairs(pairs);
    return true;
  },

  // List all pairs (owner only)
  listPairs: function(requesterNumber) {
    if (requesterNumber !== OWNER_NUMBER) return [];
    const pairs = this.loadPairs();
    return pairs.pairs;
  },

  // Generate menu for a user
  getMenu: function(userNumber) {
    const menu = `
╭━━ ◇「 ° PRIME KILLER MD BOT ° 」◇
┃⌬ BOT : ⛧ＰＲＩＭΞ⛧ ᛕΙᄂᄂΞＲ ⛧CЯΛSᕼΞЯ⛧ ɃЦ₲ ɃØŦ
┃⌬ OWNER : ${OWNER_NUMBER}
┃⌬ PLATFORM : WhatsApp
┃⌬ CONTACT : ${TELEGRAM_CONTACT}
╰━━━━━━━━━━━━━━━◇

╭━━ ◇「 COMMANDS 」◇
┃⌬ .pair <your_number> - Pair your WhatsApp number
┃⌬ .delpair - Delete all pairs (Owner only)
┃⌬ .listpair - List all paired numbers (Owner only)
┃⌬ .antidelete - Anti delete messages
┃⌬ .antilink - Anti link messages
┃⌬ .promote / .demote / .kickall - Group admin commands
┃⌬ .image / .video / .song / .tiktok / .yts - Media commands
┃⌬ .bug - Bug / Crash (COMING SOON)
┃⌬ .ping - Bot ping test
╰━━━━━━━━━━━━━━━◇

🔗 WhatsApp Channel: ${WHATSAPP_CHANNEL}
🔗 Telegram Contact: ${TELEGRAM_CONTACT}

Powered by ⛧ＰＲＩΜΞ⛧ kîᄂᄂér ⛧ƘΞИŦ⛧
`;
    return menu;
  }
};

File: utilis.py
# utils.py
import json
import os

# Pair JSON path
PAIR_FILE = "pairs.json"

# Initialize bot (example placeholder)
def init_bot(owner):
    # Replace with your actual WhatsApp bot initialization
    class Bot:
        def send_message(self, user, msg):
            print(f"[WHATSAPP MESSAGE] To {user}:\n{msg}\n")
        
        def listen(self):
            # Placeholder for listening messages
            while True:
                user = input("User number: ")
                message = input("Message: ")
                yield user, message
    return Bot()

# Return the menu text
def menu_text():
    return """
╭━━ ◇「 ⛧ＰＲＩΜΞ⛧ ᛕΙᄂᄂΞＲ ⛧CЯΛSᕼΞЯ⛧ ɃЦ₲ ɃØŦ 」◇
┃⌬ .pair - Connect device
┃⌬ .delpair - Remove device
┃⌬ .listpair - View all devices
┃⌬ .antidelete - Anti Delete
┃⌬ .antilink - Anti Link
┃⌬ .promote - Promote member
┃⌬ .demote - Demote member
┃⌬ .kickall - Kick all
┃⌬ .open - Open group
┃⌬ .close - Close group
┃⌬ .image - Download image
┃⌬ .video - Download video
┃⌬ .song - Download song
┃⌬ .tiktok - Download TikTok
┃⌬ .yts - YouTube search
┃⌬ .vcf - Download VCF
┃⌬ .bug - Bug/Crash
┃⌬ .ping - Bot Ping
╰━━━━━━━━━━━━━━
"""

# Pairing commands
def pair_command(bot, message, user):
    code = "PRIMEMD1"
    number = message.split(" ")[1] if len(message.split(" ")) > 1 else None
    if number:
        # Save paired number
        if os.path.exists(PAIR_FILE):
            with open(PAIR_FILE, "r") as f:
                pairs = json.load(f)
        else:
            pairs = {}

        pairs[number] = {"user": str(user)}
        with open(PAIR_FILE, "w") as f:
            json.dump(pairs, f, indent=4)
        bot.send_message(user, f"✅ Number {number} paired successfully with code {code}")
    else:
        bot.send_message(user, "❌ Please provide your number to pair. Example: .pair 2547xxxxxxx")

# Owner-only commands
def owner_command(bot, message, user):
    text = message.lower()
    if text.startswith(".delpair"):
        number = message.split(" ")[1] if len(message.split(" ")) > 1 else None
        if number and os.path.exists(PAIR_FILE):
            with open(PAIR_FILE, "r") as f:
                pairs = json.load(f)
            if number in pairs:
                pairs.pop(number)
                with open(PAIR_FILE, "w") as f:
                    json.dump(pairs, f, indent=4)
                bot.send_message(user, f"✅ Number {number} removed successfully.")
            else:
                bot.send_message(user, f"❌ Number {number} not found.")
        else:
            bot.send_message(user, "❌ Please provide the number to remove. Example: .delpair 2547xxxxxxx")
    
    elif text.startswith(".listpair"):
        if os.path.exists(PAIR_FILE):
            with open(PAIR_FILE, "r") as f:
                pairs = json.load(f)
            msg = "📃 Paired Numbers:\n"
            for n, data in pairs.items():
                msg += f"{n} -> {data['user']}\n"
            bot.send_message(user, msg)
        else:
            bot.send_message(user, "❌ No paired numbers found.")
